python/ray/autoscaler/node_launcher.py

```diff
@@ -36,7 +36,10 @@ def _launch_node(self, config: Dict[str, Any], count: int,
             assert node_type, node_type
         worker_filter = {TAG_RAY_NODE_KIND: NODE_KIND_WORKER}
         before = self.provider.non_terminated_nodes(tag_filters=worker_filter)
-        launch_hash = hash_launch_conf(config["worker_nodes"], config["auth"])
+        launch_hash = hash_launch_conf(
+            config["worker_nodes"],
+            config["auth"],
+            available_node_types=config.get("available_node_types"))
```

Comment:
We shouldn't include the entire config as part of the launch conf, or you need to relaunch if any config changes. Can you extract just the relevant one?

Reply:
Same comment for launch_hash in commands.py